content/blog/2019-04-25-manipulating-latex-in-rmd.md

```diff
@@ -0,0 +1,171 @@
+
+```
+file.copy(system.file("rmd/latex/default-1.17.0.2.tex",
+          package = "rmarkdown"), "template.tex")
+```
+
+If you look past the pandoc nastiness in this template file (I at least find it nasty, being that I was mostly unfamiliar with pandoc scripting!), you'll see familiar LaTeX commands that are often surrounded by `$if(X)$` statements that are triggered if `X` appears in your `.Rmd` YAML. Here's a straightforward example where, if you have `title:` in your `.Rmd` YAML header, the `\maketitle` command will be executed in your LaTeX render:
+
+![if(title)](/blog/2019-04-25-manipulating-latex-in-rmd_files/if-title.png)
+
+Now the problem feels more tractable: all we have to do is modify the `\maketitle` defaults in the usual LaTeX manner within the `template.tex` document. Let's start with the following:
+
+![Edit 1](/blog/2019-04-25-manipulating-latex-in-rmd_files/edit1.png)
+
+And don't forget to include `template.tex` in your `.Rmd` YAML header like so:
+
+```
+---
+title: "A short report"
+author: "Travis Gerke"
+date: "`r format(Sys.time(), '%Y %B %d')`"
+output: 
+   pdf_document: 
+      template: template.tex
+      keep_tex: true
+latex_engine: pdflatex
+---
+```
+
+Rendering gives the below, nice!
+
+![Edited render](/blog/2019-04-25-manipulating-latex-in-rmd_files/render2.png)
+
+Next, we clearly need to fix the fact that section titles are now larger than the document title! Let's do this with the LaTeX `sectsty` package -- you can basically stuff this code anywhere in the preamble, like so:
+
+![Adding sectsty](/blog/2019-04-25-manipulating-latex-in-rmd_files/sectsty.png)
+
+Let's also reduce the overall margins a touch via the `geometry` argument in the YAML while we're at it. Here's the full `.Rmd`:
+
+```
+---
+title: "A short report"
+author: "Travis Gerke"
+date: "`r format(Sys.time(), '%Y %B %d')`"
+output: 
+   pdf_document: 
+      template: template.tex
+      keep_tex: true
+geometry: margin=1in
+latex_engine: pdflatex
+---
+   
+# Introduction
+
+`r paste(stringi::stri_rand_lipsum(2), collapse = "\n\n")`
+
+# Progress
+
+`r paste(stringi::stri_rand_lipsum(2, start_lipsum = FALSE), collapse = "\n\n")`
+
+# Future Directions
+
+`r paste(stringi::stri_rand_lipsum(3, start_lipsum = FALSE), collapse = "\n\n")`
+```
+
+![Edited render](/blog/2019-04-25-manipulating-latex-in-rmd_files/render3.png)
+
+Mission accomplished!
+
+--- 
+
+Here's a bonus I learned along the way. You can send custom arguments to your `.tex` doc via YAML with the `params:` construction. Suppose I want an optional subtitle parameter. This is accomplished like so:
```

Comment:
It might be good to link to [the pandoc docs](https://pandoc.org/MANUAL.html#using-variables-in-templates) here.

Also, from the docs on pandoc's [yaml_metadata_block](https://pandoc.org/MANUAL.html#extension-yaml_metadata_block), any metadata variable in the YAML header is available to the template. So you don't need to use the `params:` level header, you can set `subtitle` at the YAML root level and then reference `$subtitle$` in your template (rather than `$params.subtitle$`).

```yaml
---
title: "A short report"
subtitle: "FY 2018"
---
```

In fact, it's probably best to avoid using `params:` as this has [special meaning to rmarkdown](https://bookdown.org/yihui/rmarkdown/parameterized-reports.html).